src/Adapter/Supplier/QueryHandler/GetSupplierForViewingHandler.php

```diff
@@ -0,0 +1,174 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Supplier\QueryHandler;
+
+use Context;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Exception\SupplierException;
+use PrestaShop\PrestaShop\Core\Localization\Exception\LocalizationException;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
+use PrestaShopException;
+use Supplier;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Exception\SupplierNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Query\GetSupplierForViewing;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\QueryResult\ViewableSupplier;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\ValueObject\SupplierId;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\QueryHandler\GetSupplierForViewingHandlerInterface;
+use Product;
+
+/**
+ * Handles query which gets supplier for viewing
+ */
+class GetSupplierForViewingHandler implements GetSupplierForViewingHandlerInterface
+{
+    /**
+     * @var LocaleRepository
+     */
+    private $localeRepository;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param LocaleRepository $localeRepository
+     * @param Context $context
+     */
+    public function __construct(LocaleRepository $localeRepository, Context $context)
+    {
+        $this->localeRepository = $localeRepository;
+        $this->context = $context;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws SupplierException
+     * @throws LocalizationException
+     */
+    public function handle(GetSupplierForViewing $query)
+    {
+        $supplier = $this->getSupplier($query->getSupplierId());
+
+        return new ViewableSupplier(
+            $supplier->name,
+            $this->getSupplierProducts($supplier, $query->getLanguageId())
+        );
+    }
+
+    /**
+     * @param SupplierId $supplierId
+     *
+     * @return Supplier
+     *
+     * @throws SupplierNotFoundException
+     */
+    private function getSupplier(SupplierId $supplierId)
+    {
+        $supplier = new Supplier($supplierId->getValue());
+
+        if ($supplier->id !== $supplierId->getValue()) {
+            throw new SupplierNotFoundException(
+                sprintf('Supplier with id "%s" was not found.', $supplierId->getValue())
+            );
+        }
+
+        return $supplier;
+    }
+
+    /**
+     * @param Supplier $supplier
+     * @param LanguageId $languageId
+     *
+     * @return array
+     *
+     * @throws LocalizationException
+     * @throws SupplierException
+     */
+    private function getSupplierProducts(Supplier $supplier, LanguageId $languageId)
+    {
+        $products = [];
+        $supplierProducts = $supplier->getProductsLite($languageId->getValue());
+        $locale = $this->localeRepository->getLocale($this->context->language->getLocale());
```

Comment:
I'm wondering if `locale` should come from `context` or `query`. :thinking:

Reply:
Not sure either, @matks, wdyt?

Reply:
Let's stick to legacy behavior (for now)